pypdf/generic/_base.py

```diff
@@ -286,6 +286,16 @@ def get_object(self) -> Optional["PdfObject"]:
             return None
         return obj.get_object()
 
+    @staticmethod
+    def fully_unwrap(obj: Optional["PdfObject"]) -> Optional["PdfObject"]:
```

Comment:
I do not really like such recursive calls which lead to recursion errors for malicious files (see #2251). Thus we might want to introduce some limit here?

Reply:
Well spotted, I'll fix this up.